src/AI/ProviderFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BumbleDocGen\AI;
+
+use BumbleDocGen\AI\Providers\HuggingFace\Provider as HuggingFaceProvider;
+use BumbleDocGen\AI\Providers\OpenAI\Provider as OpenAIProvider;
+use BumbleDocGen\AI\Providers\Ollama\Provider as OllamaProvider;
+use RuntimeException;
+
+final class ProviderFactory
+{
+    public static function create(): ProviderInterface
+    {
+        $apiType = getenv('API_TYPE');
+        switch ($apiType) {
+            case 'huggingface':
+                $endpoint = getenv('HUGGINGFACE_ENDPOINT');
+                $apiKey = getenv('HUGGINGFACE_API_KEY');
+                if (empty($endpoint)) {
+                    throw new RuntimeException("Environment variable HUGGINGFACE_ENDPOINT not set!");
+                }
+                if (empty($apiKey)) {
+                    throw new RuntimeException("Environment variable HUGGINGFACE_TOKEN not set!");
+                }
+                return new HuggingFaceProvider($apiKey, $endpoint);
+            case 'ollama':
+                return new OllamaProvider();
+            case 'openai':
+                $apiKey = getenv('OPENAI_API_KEY');
+                if (empty($apiKey)) {
+                    throw new RuntimeException("Environment variable OPENAI_GPT_MODEL not set!");
+                }
+                return new OpenAIProvider($apiKey);
+            default:
+                throw new RuntimeException(
+                    "Environment variable API_TYPE not set to valid option (openai, huggingface)!",
```

Comment:
We have more options, let's generate this message based on existing providers dynamically